service/dap/server.go

```diff
@@ -1549,13 +1553,80 @@ type stackFrame struct {
 // As per DAP spec, this request only gets triggered as a follow-up
 // to a successful threads request as part of the "request waterfall".
 func (s *Server) onStackTraceRequest(request *dap.StackTraceRequest) {
+	// In order to support paging of stack frames, we always check an additional 10
+	// beyond the number of stack frames that were requested. Since the total frames
+	// will be larger than the number of frames loaded, the client will continue to
+	// request them.
 	goroutineID := request.Arguments.ThreadId
-	frames, err := s.debugger.Stacktrace(goroutineID, s.args.stackTraceDepth, 0)
+	start := request.Arguments.StartFrame
+	if start < 0 {
+		start = 0
+	}
+	var levels int
+	if request.Arguments.Levels > 0 {
+		levels = request.Arguments.Levels
+	} else {
+		levels = s.args.stackTraceDepth
+	}
+
+	stackFrames, totalFrames, hasMore, err := s.loadFrames(goroutineID, start, levels)
 	if err != nil {
 		s.sendErrorResponse(request.Request, UnableToProduceStackTrace, "Unable to produce stack trace", err.Error())
 		return
 	}
 
+	if hasMore {
+		totalFrames += 20
+	}
+	response := &dap.StackTraceResponse{
+		Response: *newResponse(request.Request),
+		Body:     dap.StackTraceResponseBody{StackFrames: stackFrames, TotalFrames: totalFrames},
+	}
+	s.send(response)
+}
+
+func (s *Server) loadFrames(goroutineID, start, levels int) ([]dap.StackFrame, int, bool, error) {
```

Comment:
A function comment would be helpful. For example, is the returned int - total loaded frames (which would be either levels or less) or total frames in the entire stacktrace, including that are yet to be loaded?